worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala

```diff
@@ -435,6 +435,14 @@ private[celeborn] class Worker(
       0
     }
   }
+  // Unreleased shuffle count when worker was decommissioned
+  workerSource.addGauge(WorkerSource.UNRELEASED_SHUFFLE_COUNT) { () =>
+    if (shutdown.get() && workerStatusManager.currentWorkerStatus.getState == State.Exit) {
```

Comment:
```suggestion
    if (shutdown.get() && workerStatusManager.currentWorkerStatus.getState == State.InDecommission ||
        workerStatusManager.currentWorkerStatus.getState == State.InDecommissionThenIdle) {
```
The status should be `InDecommission` or `InDecommissionThenIdle`. Right? Because the description of this metric is that `Unreleased shuffle count when worker was decommissioned. `.

Reply:
After transitioning to `InDecommission` state it will wait for `workerDecommissionForceExitTimeout` to cleanup the shuffle. My thought is that we should only check the count of shuffle which was not able to get released till the timeout.

Suggested condition will mean "Unreleased shuffle count when worker started decommissioning".

Reply:
@s0nskar, `Unreleased shuffle count when worker started decommissioning` looks good. You could update the description of the metric.

Reply:
Yeah that also make sense. Let me update the PR.